### PR TITLE
Improve scoping of two potentially overlapping Triage sections

### PR DESCRIPTION
**Pull request by benjaoming**
* [x] Improve the scope of the two Triaging sections
* [x] Fixes wrong ``:doc:`team` `` referencing Write the Docs via intersphinx
* [x] ~This is a bit meta-level: Since I'm going over dev docs right now, I wonder how exactly this PR lands: It's a PR without an issue, so it's not actually triaged. Some projects prefer that an issue is always opened for discussion, some prefer that contributors have a shortcut by simply opening a PR.~ I believe that this is resolved, we keep things as they are, adding a bit more incentives on the triaging process.
* [x] Do not allow any `:doc:` or `:ref:` to resolve to intersphinx mapping without `:external:` prefix
* [x] Ignore `:external:` role in rstcheck

### CODE REVIEW

File: docs/dev/contribute.rst
Comment (stsewd) [inline docs/dev/contribute.rst:96]: We prefer to use absolute paths (in case pages are moved)

```suggestion
issues on Github. The Read the Docs core :doc:`/team` uses the following
```

Comment (benjaoming) [reply docs/dev/contribute.rst:96]: ``:doc:`/team` `` breaks correctly (because there is no such document)

There is actually no document `/team` and Intersphinx was automatically converting ``:doc:`team` `` to https://www.writethedocs.org/team/ -- I copy-pasted without looking at where it was pointing. I'll remove this reference from the original `note::`.

It seems like a good idea to use absolute paths. In case `/dev/teams.rst` has existed (I don't see it in the git repo history though), removing the file would also break the build, which is what we want.

Comment (humitos) [reply docs/dev/contribute.rst]: We should probably use this config https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html#confval-intersphinx_disabled_reftypes to avoid this happening again.

File: docs/dev/contribute.rst
Comment (benjaoming) [reply docs/dev/contribute.rst]: This is fixed, thanks! I think I understand how to use it -- when changing a section title, we rely on the documentation build to fail - so it's all the same as if explicit section labels were used?

Comment (benjaoming) [reply docs/dev/contribute.rst]: I guess the only catch is that we cannot rename a section without also changing all the references?

File: docs/dev/contribute.rst
Comment (humitos) [inline docs/dev/contribute.rst:93]: Is this needed here? I saw it in the `.rstcheck.cfg` file as well.

File: docs/dev/roadmap.rst
Comment (humitos) [reply docs/dev/roadmap.rst:27]: It seems this is not applied, right?

File: docs/dev/contribute.rst
Comment (humitos) [reply docs/dev/contribute.rst]: Yes, that's kind of a downside, but keeps things more explicit.

Comment (stsewd) [reply docs/dev/contribute.rst]: We have a small guide about this at https://docs.readthedocs.io/en/stable/guides/cross-referencing-with-sphinx.html#automatically-label-sections if it's useful :)

File: docs/dev/contribute.rst
Comment (benjaoming) [reply docs/dev/contribute.rst:93]: Sorry no, this is an artifact from a previous attempt :)

File: dockerfiles/Dockerfile
Comment (humitos) [inline dockerfiles/Dockerfile:39]: It seems you included this change in this PR by mistake.

Comment (benjaoming) [reply dockerfiles/Dockerfile]: Oh yeah, that wasn't supposed to have happened... CTRL-C'ed the git push process too late :)

**Review by humitos — APPROVED** [review]
Looks good to me 👍🏼 

One comment. Please, do not force push new commits if there is a review already done in the PR because that forces the reviewers to review all the files again since GitHub interprets as all the content is new. If there is a way to solve this problem involving force push I'm happy to know how, but I didn't find it. If you want to rebase/change history/etc you can do that immediately before merging the PR

**Comment by benjaoming** [comment]
> Please, do not force push new commits if there is a review already done in the PR because

This is done so that - as a PR Author - can ensure that the commit history is nice and clean before a merge. There are several alternatives to the same goal:

1) A maintainer asks the PR Author to rewrite the commit history
2) A maintainer chooses squash/rebase and the branch is rewritten before a merge -- but without any manual choices or interventions
3) Maintainer uses the `!fixup` notation from the beginning -- I think in this case, it will be quite explicit for the maintainer that a squash is expected before a final merge. Github.com doesn't support anything out of the box for this yet, so it's the same process as 1) AFAIK.

Of course, there is also the option of keeping all the commits and merging the old way.

> that forces the reviewers to review all the files again since GitHub interprets as all the content is new.

This is not really true, you can see all comments and what segments of the code they belong to. All comments are also outdated.

Personally, I like to take a pragmatic approach and not see any of these as a mandatory option, as it raises the bar for contribution and makes it too easy for developers to have discussions :smile:

**Comment by humitos** [comment]
> > that forces the reviewers to review all the files again since GitHub interprets as all the content is new.
> 
> This is not really true, you can see all comments and what segments of the code they belong to. All comments in this PR were outdated.

I'm not talking about the comments, but the changes themselves. If you force push, when I go to "File changed" tab, GitHub shows me _all the files_ again, instead of only the last changes since I reviewed this PR last time. So, I lost all the files I marked as "Viewed" and I'm not sure what changes from the last time I review the PR.

**Pull request merged**
